src/Catty/FormulaEditor/ViewController/FormulaEditorSectionViewController.swift

```diff
@@ -0,0 +1,405 @@
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+@objc enum FormulaEditorSectionType: Int {
+    case none
+    case functions
+    case object
+    case logic
+    case sensors
+    case data
+}
+
+@objc class FormulaEditorSectionViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+
+    private var placeHolderLabel = UILabel()
+    private var addButton = UIBarButtonItem()
+    @objc var formulaEditorVC: FormulaEditorViewController?
+    @objc var formulaEditorSectionType: FormulaEditorSectionType = .none
+    @objc var formulaManager: FormulaManager?
+    @objc var spriteObject: SpriteObject?
+
+    private var items = [FormulaEditorItem]()
+    private var numberOfSections = 0
+    private var numberOfRowsInSection = [Int]()
+    private var titlesOfSections = [String]()
+
+    var variableSourceProject = [UserVariable]()
+    var variableSourceObject = [UserVariable]()
+    var listSourceProject = [UserList]()
+    var listSourceObject = [UserList]()
+    var newVarIsForProject = false
+
+    private struct VariableOrList {
+        let name: String
+        let isList: Bool
+        let projectScope: Bool
+    }
+
+    private var allVariablesAndLists = [VariableOrList]()
+
+    private var tableView = UITableView()
+
+    override func viewDidLoad() {
+
+        self.view.backgroundColor = .white
+
+        self.addButton = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(self.addButtonTapped))
+        self.addButton.tintColor = .light
+
+        let tableViewTopConstraint = NSLayoutConstraint(item: self.tableView, attribute: .top, relatedBy: .equal, toItem: self.view, attribute: .top, multiplier: 1, constant: 0)
+        let tableViewBottomConstraint = NSLayoutConstraint(item: self.tableView, attribute: .bottom, relatedBy: .equal, toItem: self.view, attribute: .bottom, multiplier: 1, constant: 0)
+        let tableViewLeadingConstraint = NSLayoutConstraint(item: self.tableView, attribute: .leading, relatedBy: .equal, toItem: self.view, attribute: .leading, multiplier: 1, constant: 0)
+        let tableViewTrailingConstraint = NSLayoutConstraint(item: self.tableView, attribute: .trailing, relatedBy: .equal, toItem: self.view, attribute: .trailing, multiplier: 1, constant: 0)
+        self.tableView.translatesAutoresizingMaskIntoConstraints = false
+        self.tableView.dataSource = self
+        self.tableView.delegate = self
+        self.view.addSubview(self.tableView)
+        self.view.addConstraints([tableViewTopConstraint, tableViewBottomConstraint, tableViewLeadingConstraint, tableViewTrailingConstraint])
+
+        self.placeHolderLabel = UILabel(frame: CGRect(width: self.view.frame.width, height: 45))
+        self.placeHolderLabel.text = kUIFEAddVariablesAndLists
```

Comment:
Can you please make sure that this text is also fully visible on smaller devices (e.g., iPhone SE models). Like on the Object-screen it should be then displayed in multiple lines (in the case the screen is too small):

Current situation:
![Data](https://user-images.githubusercontent.com/6848980/108172320-fdf9ad00-70fc-11eb-89b4-127e6cfb1020.png)

Desired situation:
![Objects](https://user-images.githubusercontent.com/6848980/108172326-018d3400-70fd-11eb-8870-918541067802.png)